apps/api/src/app/step-schemas/e2e/get-step-schema.e2e.ts

```diff
@@ -0,0 +1,283 @@
+import { expect } from 'chai';
+
+import { UserSession } from '@novu/testing';
+import { StepTypeEnum, WorkflowResponseDto } from '@novu/shared';
+
+describe('Get Step Schema - /step-schemas?workflowId=:workflowId&stepId=:stepId&stepType=:stepType (GET)', async () => {
+  let session: UserSession;
+  let createdWorkflow: WorkflowResponseDto;
+
+  beforeEach(async () => {
+    // @ts-ignore
+    process.env.IS_WORKFLOW_PREFERENCES_ENABLED = 'true';
+    session = new UserSession();
+    await session.initialize();
+
+    const workflowObject = {
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+      name: 'test api template',
+      description: 'This is a test description',
+      tags: ['test-tag-api'],
+      notificationGroupId: session.notificationGroups[0]._id,
+      steps: [
+        {
+          name: 'In-App Test Step',
+          type: StepTypeEnum.IN_APP,
+        },
+        {
+          name: 'SMS Test Step',
+          type: StepTypeEnum.SMS,
+        },
+      ],
+      triggers: [{ identifier: 'test-trigger-api' }],
+    };
+
+    const workflowDataRes = await session.testAgent.post(`/v2/workflows`).send(workflowObject);
+    createdWorkflow = workflowDataRes.body.data;
+  });
+
+  afterEach(async () => {
+    // @ts-ignore
+    process.env.IS_WORKFLOW_PREFERENCES_ENABLED = 'false';
+  });
+
+  describe('Get Control Schema with stepType', () => {
+    it('should get step schema for in app step type', async function () {
+      const { data } = (await session.testAgent.get(`/v1/step-schemas?stepType=${StepTypeEnum.IN_APP}`)).body;
+
+      // in app output schema
+      expect(data.controls.type).to.equal('object');
+      expect(data.controls.properties).to.have.property('subject');
+      expect(data.controls.properties.subject.type).to.equal('string');
+      expect(data.controls.properties).to.have.property('body');
+      expect(data.controls.properties.body.type).to.equal('string');
```

Comment:
￼This seems quite specific. Perhaps we could consider building an object and implementing a deep match instead? It might help improve maintainability in the long run.

Reply:
I do like more specific tests, for example, let's look at the `origin` bug we had today.